test/options.ts

```diff
@@ -0,0 +1,138 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as assert from 'assert';
+import {resolve} from 'path';
+import {parseOptions, FrameworkOptions} from '../src/options';
+
+describe('parseOptions', () => {
+  interface TestData {
+    name: string;
+    cliOpts: string[];
+    envVars: {[key: string]: string};
+    expectedOptions: Partial<FrameworkOptions>;
+  }
+
+  const testData: TestData[] = [
+    {
+      name: 'parses the cli --help option',
+      cliOpts: ['bin/node', '/index.js', '--help'],
+      envVars: {},
+      expectedOptions: {
+        printHelp: true,
+      },
+    },
+    {
+      name: 'parses the cli -h option',
+      cliOpts: ['bin/node', '/index.js', '-h'],
+      envVars: {},
+      expectedOptions: {
+        printHelp: true,
+      },
+    },
+    {
+      name: 'sets the correct defaults for all options',
+      cliOpts: ['bin/node', '/index.js'],
+      envVars: {},
+      expectedOptions: {
+        port: '8080',
+        target: 'function',
+        sourceLocation: resolve(''),
+        signatureType: 'http',
+        printHelp: false,
+      },
+    },
+    {
+      name: 'respects all cli flags',
+      cliOpts: [
+        'bin/node',
+        '/index.js',
+        '--port',
+        '1234',
+        '--target=helloWorld',
+        '--signature-type',
+        'cloudevent',
+        '--source=/source',
```

Comment:
nit: we're mixing using ` ` and `=` for arguments. Can we choose one (`=` seems cleaner)? Here and elsewhere.

Reply:
This was intentional to make sure we get test coverage for both syntax options. Alternatively I could have split this into two tests cases for `--this case` and `--this=case`. But I thought this was a bit more terse.